examples/chip-tool/commands/tests/Commands.h

```diff
@@ -7829,6 +7832,75 @@ class TestCluster : public TestCommand
 
         runner->NextTest();
     }
+
+    // Test Send Test Command to unsupported endpoint
+    typedef void (*SuccessCallback_101)(void * context);
+    chip::Callback::Callback<SuccessCallback_101> * mOnSuccessCallback_101    = nullptr;
+    chip::Callback::Callback<DefaultFailureCallback> * mOnFailureCallback_101 = nullptr;
+    bool mIsFailureExpected_101                                               = 1;
+
+    CHIP_ERROR TestSendClusterTestClusterCommandTest_101()
+    {
+        ChipLogProgress(chipTool, "Test Cluster - Send Test Command to unsupported endpoint: Sending command...");
+
+        mOnFailureCallback_101 =
+            new chip::Callback::Callback<DefaultFailureCallback>(OnTestSendClusterTestClusterCommandTest_101_FailureResponse, this);
+        mOnSuccessCallback_101 =
+            new chip::Callback::Callback<SuccessCallback_101>(OnTestSendClusterTestClusterCommandTest_101_SuccessResponse, this);
+
+        chip::Controller::TestClusterCluster cluster;
+        cluster.Associate(mDevice, 200);
+
+        CHIP_ERROR err = CHIP_NO_ERROR;
+
+        err = cluster.Test(mOnSuccessCallback_101->Cancel(), mOnFailureCallback_101->Cancel());
+
+        if (CHIP_NO_ERROR != err)
+        {
+            delete mOnFailureCallback_101;
+            delete mOnSuccessCallback_101;
+        }
+
+        return err;
+    }
+
+    static void OnTestSendClusterTestClusterCommandTest_101_FailureResponse(void * context, uint8_t status)
+    {
+        ChipLogProgress(chipTool, "Test Cluster - Send Test Command to unsupported endpoint: Failure Response");
+
+        TestCluster * runner = reinterpret_cast<TestCluster *>(context);
+
+        delete runner->mOnFailureCallback_101;
+        delete runner->mOnSuccessCallback_101;
+
+        if (runner->mIsFailureExpected_101 == false)
+        {
+            ChipLogError(chipTool, "Error: The test was expecting a success callback. Got failure callback");
+            runner->SetCommandExitStatus(CHIP_ERROR_INTERNAL);
+            return;
+        }
+
+        runner->NextTest();
+    }
+
+    static void OnTestSendClusterTestClusterCommandTest_101_SuccessResponse(void * context)
```

Comment:
Looks like this pattern repeats for other test cases as well. Could this callback functions be common for all of them? Is there a reason we can't have a common `runner->mIsFailureExpected` member?

Reply:
Possibly.  This is auto-generated code; I'd have to go look at the template to see whether we could do that and whether the resulting template would be more complicated or not as a result....

Reply:
The template has various conditions in it; depending on the exact way the test is configured in the yaml you get different signatures.  So commoning up the functions is not really very simple.

Similarly, `mIsFailureExpected` is on a per-test basis, and the runner is a single runner across a suite of tests, so we can't just have one member on it to track the per-test state.